Src/VimCompletion/Implementation/WordCompletion/Async/WordAsyncCompletionSession.cs

```diff
@@ -0,0 +1,128 @@
+using Vim;
+using System.Threading;
+using System.Windows.Threading;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.Editor;
+using System.Reflection;
+
+namespace Vim.Implementation.WordCompletion.Async
+{
+    /// <summary>
+    /// Implementation of the IWordCompletionSession interface.  
+    /// to provide the friendly interface the core Vim engine is expecting
+    /// </summary>
+    internal sealed class WordAsyncCompletionSession : IWordCompletionSession
+    {
+        private readonly ITextView _textView;
+        private readonly IAsyncCompletionSession _asyncCompletionSession;
+        private bool _isDismissed;
+        private event EventHandler _dismissed;
+        private readonly DispatcherTimer _tipTimer;
+        private readonly IVsTextView _vsTextView;
+
+        internal WordAsyncCompletionSession(IAsyncCompletionSession asyncCompletionSession, IVsEditorAdaptersFactoryService vsEditorAdaptersFactoryService = null)
+        {
+            _textView = asyncCompletionSession.TextView;
+            _asyncCompletionSession = asyncCompletionSession;
+            _asyncCompletionSession.Dismissed += delegate { OnDismissed(); };
+            _vsTextView = vsEditorAdaptersFactoryService?.GetViewAdapter(_textView);
+            if (_vsTextView is object)
+            {
+                _tipTimer = new DispatcherTimer(TimeSpan.FromMilliseconds(250), DispatcherPriority.Normal, callback: ResetTipOpacity, Dispatcher.CurrentDispatcher);
+                _tipTimer.Start();
+            }
+        }
+
+        /// <summary>
+        /// Called when the session is dismissed.  Need to alert any consumers that we have been
+        /// dismissed 
+        /// </summary>
+        private void OnDismissed()
+        {
+            _isDismissed = true;
+            _textView.ClearWordCompletionData();
+            _dismissed?.Invoke(this, EventArgs.Empty);
+            _tipTimer?.Stop();
+        }
+
+        private bool WithOperations(Action<IAsyncCompletionSessionOperations> action)
+        {
+            if (_asyncCompletionSession is IAsyncCompletionSessionOperations operations)
+            {
+                action(operations);
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool MoveDown() => WithOperations(operations => operations.SelectDown());
+        private bool MoveUp() => WithOperations(operations => operations.SelectUp());
+        private void Commit() => _asyncCompletionSession.Commit(KeyInputUtil.EnterKey.Char, CancellationToken.None);
+
+        /// <summary>
+        /// The async completion presenter will fade out the completion menu when the control key is clicked. That 
+        /// is unfortunate for VsVim as control is held down for the duration of a completion session. This ... method
+        /// is used to reset the opacity to 1.0 
+        /// </summary>
+        private void ResetTipOpacity(object sender, EventArgs e)

```

Comment:
Does this mean that the opacity fades a little over the course of 250 msec in a triangle wave pattern? Is it noticeable as a subtle "pulsing"? I will build and test the branch to see if I can notice it. This is different than the way that the old completion session worked around the opacity problem.

Reply:
I'm thinking the opacity does this:

```
\    |\    |\
 \   | \   | \
  \  |  \  |  \
   \ |   \ |   \
    \|    \|    \
```

Reply:
In general it should not do this. The implementation of `SetOpacity` stops the `DispatchTimer` they are using to implement the fade out effect. The reason we're running this on a timer though is that every time a VsVim developer presses control, say if they  take hands off keyboard and then hit ctrl-n again, then the timer will start anew. 

So the subsequent calls to `SetOpacity` should be no-ops in the majority case.

Reply:
Note: the specific method here is `SimpleTextViewWindow.SetOpacity` and it's in the MS.VS.Editor.Implementation.dll. That should give you a better idea of how this is working under the hood.